sql/src/main/java/io/crate/rest/action/RestResultSetReceiver.java

```diff
@@ -88,11 +88,19 @@ public void setNextRow(Row row) {
 
     @Override
     public void allFinished(boolean interrupted) {
+        BytesRestResponse response;
         try {
-            channel.sendResponse(new BytesRestResponse(RestStatus.OK, finishBuilder()));
+            response = new BytesRestResponse(RestStatus.OK, finishBuilder());
+        } catch (Throwable t) {
+            fail(t);
+            return;
+        }
+
+        try {
+            channel.sendResponse(response);
```

Comment:
In this case I think it's okay. 
In the postgres case the `super.allFinished()` call triggers a `ReadyForQuery` message that is sent to the client - I had to change that to make sure that that is only done after the `CommandComplete` message arrived on the client. In the http case there is only a single response.